File: src/Orchestrator/src/Application/General/Queries/GetContractsStates/VM/UserVm.cs
namespace Application.General.Queries.GetContractsStates.VM;

public class UserVm
{
    public required string Id { get; init; }
    public required string WalletAddress { get; init; }
    public required string HexBalance { get; init; }
    public required string HexStakedBalance { get; init; }
}

File: src/frontend/lib/admin/models/vm/user_vm.dart
class UserVm {
  final String id;
  final String walletAddress;
  final BigInt balance;
  final BigInt stakedBalance;

  UserVm.fromMap(Map<String, dynamic> map)
      : id = map['id'],
        walletAddress = map['walletAddress'],
        balance = BigInt.parse(map['hexBalance']),
        stakedBalance = BigInt.parse(map['hexStakedBalance']);
}
